src/contexts/WebSocketContext.tsx

```
@@ -0,0 +1,188 @@
+      };
+      
+      socket.onclose = (event) => {
+        console.log(`WebSocket connection closed: ${event.code} ${event.reason}`);
+        setConnected(false);
+        
+        // Don't try to reconnect if we've explicitly closed the connection (code 1000)
+        if (event.code !== 1000 && reconnectAttemptRef.current < maxReconnectAttempts) {
+          attemptReconnect();
+        }
+      };
+    } catch (error) {
```

ESLint

[error] 'error' is already declared in the upper scope on line 38 column 10.

(@typescript-eslint/no-shadow)